### PR TITLE
perf: less collision - less comparison

### PR DESCRIPTION
**Pull request by obastemur**
AcmeAIR LTO gain ~2%

- Do not CrossSite::MarshalVar if scriptContext == requestContext
   Saves 3 compare + arithmetic ops per each change.

 - Use prime policy for inline cache dictionaries
    Less collision rate, more responsive

### CODE REVIEW

File: lib/Runtime/Language/JavascriptOperators.cpp [hunk 1]
```diff
@@ -9367,7 +9370,12 @@ namespace Js
 
             Var thisVar = RootToThisObject(object, scriptContext);
 
-            RecyclableObject* marshalledFunction = RecyclableObject::FromVar(CrossSite::MarshalVar(requestContext, function));
+            RecyclableObject* marshalledFunction = function;
+            if (scriptContext != requestContext)
```
Comment (leirocks) [inline lib/Runtime/Language/JavascriptOperators.cpp]: can you make an overload of CrossSite::MarshalVar() to take both scriptContext as input parameter, to make the caller code cleaner

Comment (obastemur) [reply lib/Runtime/Language/JavascriptOperators.cpp]: @leirocks good point. I was thinking about it, let me update the PR.

Comment (tcare) [reply lib/Runtime/Language/JavascriptOperators.cpp]: +1, I like that convention.

File: lib/Runtime/Types/TypePropertyCache.cpp [hunk 2]
```diff
@@ -296,8 +299,10 @@ namespace Js
                 requestContext);
             return true;
         }
-
-        *propertyValue = CrossSite::MarshalVar(requestContext, *propertyValue);
+        else
+        {
+            *propertyValue = CrossSite::MarshalVar(requestContext, *propertyValue);
```
Comment (tcare) [inline lib/Runtime/Types/TypePropertyCache.cpp]: Do we need to marshal here? We already had this check before and didn't marshal in the false case..

Comment (obastemur) [reply lib/Runtime/Types/TypePropertyCache.cpp]: code reaches here only when `! if(prototypeObjectWithProperty->GetScriptContext() == requestContext)`

Comment (tcare) [reply lib/Runtime/Types/TypePropertyCache.cpp]: Right, but it already did before without marshalling. Is there an explicit need now?

Comment (obastemur) [reply lib/Runtime/Types/TypePropertyCache.cpp]: Not sure. I just added `else` there.

Comment (tcare) [reply lib/Runtime/Types/TypePropertyCache.cpp]: Oh sorry the diff wasn't showing properly. That's fine.

**Comment by obastemur** [comment]
@dotnet-bot test Ubuntu static_ubuntu_linux_test please

**Pull request merged**